### PR TITLE
[CMSP-12] Multisite search-replace EA

### PR DESCRIPTION
**Pull request by pwtyler**
## Summary

**[WordPress Multisite Search Replace](https://docs.pantheon.io/guides/multisite/search-replace)** - Adds details for enabling search-replace on WPMS, and updates relevant sections to link back to the new page.

## Effect

The following changes are already committed:

* Adds a new page for WPMS search-replace
* Updates pantheon.yml page to point back to the new document
* Updates Multisite workflows to point to the new document

## Remaining Work and Prerequisites

- [x]  clean up the linked gist and/or embed
- [x]  Update other sections that details WPMS limitations to link back as needed
- [ ]  general wordsmithing 

### Dependencies and Timing

**Release**:
- [x] When ready

--------------------------------------------------

## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Redirect `/old-path/` => `/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] For Heroes - add a props post to the [discussion board](https://discuss.pantheon.io/c/pantheon-platform/documentation/17).
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

File: source/content/guides/multisite/05-workflows.md
Comment (scottbuscemi) [inline source/content/guides/multisite/05-workflows.md:44]: Search & Replace
Multisite Search & Replace

File: source/content/guides/multisite/07-search-replace.md
Comment (scottbuscemi) [inline source/content/guides/multisite/07-search-replace.md:21]: Search & Replace

File: source/content/guides/multisite/07-search-replace.md
Comment (scottbuscemi) [inline source/content/guides/multisite/07-search-replace.md:47]: 25

**Comment by scottbuscemi** [comment]
Just needs a mention of when s&r is run (creating new environment, deploying commits to an environment, cloning content, etc)

File: source/content/guides/multisite/07-search-replace.md
Comment (pwtyler) [inline source/content/guides/multisite/07-search-replace.md:43]: @jazzsequence, copying your comment here:
> in that case I would make sure there’s adequate info in a gh repo somewhere that the docs can link out to because there’s some nuance about how it works that should be explained more fully imo

There won't be a github repo for now, what nuance would you like clarified here?

File: source/content/guides/multisite/05-workflows.md
Comment (pwtyler) [reply source/content/guides/multisite/05-workflows.md:44]:
```suggestion
Multisite search and replace can now be handled automatically platform search & replace, currently in Early Access. For more details, see [Multisite Search Replace](/guides/multisite/search-replace).
```

File: source/content/guides/multisite/07-search-replace.md
Comment (pwtyler) [inline source/content/guides/multisite/07-search-replace.md:3]:
```suggestion
subtitle: Search & Replace
```

File: source/content/guides/multisite/07-search-replace.md
Comment (pwtyler) [inline source/content/guides/multisite/07-search-replace.md:4]:
```suggestion
description: Additional handing required for running search & replace on WordPress Multisites.
```

File: source/content/guides/multisite/07-search-replace.md
Comment (pwtyler) [reply source/content/guides/multisite/07-search-replace.md:21]:
```suggestion
WordPress Multisite (WPMS) search & replace is available for [Early Access](/oss-support-levels#early-access) participants. Features for WPMS search & replace are in active development. Pantheon's development team is rolling out new functionality often while this product is in Early Access. Visit [#wordpress in our community Slack](https://pantheon-community.slack.com/archives/CT8MC5Y0K) (you can sign up for the [Pantheon Slack channel here](https://slackin.pantheon.io/) if you don't already have an account) to learn how you can enroll in our Early Access program. Please review Pantheon's [Software Evaluation Licensing Terms](https://legal.pantheon.io/#contract-hkqlbwpxo) for more information about access to our software.
```

File: source/content/guides/multisite/07-search-replace.md
Comment (pwtyler) [inline source/content/guides/multisite/07-search-replace.md:25]:
```suggestion
This section provides information on how to configure automatic platform search-replace of URLs for both subdirectory and subdomain WordPress Multisites.
```

File: source/content/guides/multisite/07-search-replace.md
Comment (pwtyler) [reply source/content/guides/multisite/07-search-replace.md:47]:
```suggestion
There is currently a limit of 25 sites that can be configured for search-replace. We expect this number to increase before general availability.
```

File: source/content/guides/multisite/05-workflows.md
Comment (jspellman814) [inline source/content/guides/multisite/05-workflows.md:44]:
```suggestion
Multisite search and replace can now be handled automatically by platform search & replace, currently in Early Access. For more details, see [Multisite Search Replace](/guides/multisite/search-replace).
```

File: source/content/guides/multisite/07-search-replace.md
Comment (jspellman814) [inline source/content/guides/multisite/07-search-replace.md:21]:
```suggestion
WordPress Multisite (WPMS) search & replace is available for [Early Access](/oss-support-levels#early-access) participants. Features for platform search-replace are in active development. Pantheon's development team is rolling out new functionality often while this product is in Early Access. Visit [#wordpress in our community Slack](https://pantheon-community.slack.com/archives/CT8MC5Y0K) (you can sign up for the [Pantheon Slack channel here](https://slackin.pantheon.io/) if you don't already have an account) to learn how you can enroll in our Early Access program. Please review Pantheon's [Software Evaluation Licensing Terms](https://legal.pantheon.io/#contract-hkqlbwpxo) for more information about access to our software.
```

This is referred to as `platform search-replace` further down in this document - we should decide on `WPMS search-replace` or `platform search-replace` and be consistent throughout the doc.

File: source/content/guides/multisite/07-search-replace.md
Comment (jspellman814) [inline source/content/guides/multisite/07-search-replace.md:48]:
```suggestion
If search-replace is enabled for an environment, but `sites.yml` does not exist, nothing will be updated. if `sites.yml` is different between environments, the `domain_maps` in the target environment’s `sites.yml` will be used to determine what to replace.
```

File: source/content/pantheon-yml.md
Comment (jspellman814) [inline source/content/pantheon-yml.md:258]:
```suggestion
You can opt-in to platform search-replace on [WordPress Multisites](/guides/multisite) by adding the following to `pantheon.yml`:
```

Should we prepend `search-replace` here?

**Review by jspellman814 — CHANGES_REQUESTED** [review]
few small comments/questions... looks great overall though!

File: source/content/guides/multisite/07-search-replace.md
Comment (pwtyler) [reply source/content/guides/multisite/07-search-replace.md:31]: search-replace was what I started with, but search & replace was @scottbuscemi preference in a previous review; I was not comprehensive enough in my followup throughout the document.

File: source/content/guides/multisite/07-search-replace.md
Comment (pwtyler) [reply source/content/guides/multisite/07-search-replace.md:21]: re: WPMS search & replace vs platform search & replace— trying to differentiate as this behaves somewhat differently than _any s&r_ on the platform, and you can't *opt-in/out* of s&r on a non-WPMS site so that seemed confusing.

File: source/content/pantheon-yml.md
Comment (pwtyler) [reply source/content/pantheon-yml.md:258]:
```suggestion
You can opt-in to platform search & replace on [WordPress Multisites](/guides/multisite) by adding the following to `pantheon.yml`:
```

File: source/content/guides/multisite/07-search-replace.md
Comment (jazzsequence) [reply source/content/guides/multisite/07-search-replace.md:46]: Discussion [here](https://pantheon.slack.com/archives/C0397K11QEQ/p1680628327865289).

tl:dr; Broadly I'd like a description of the `sites.yml` file linked as well as information and troubleshooting steps about what validation errors might trigger and how to solve them linked in here somewhere, in addition to an example valid and annotated `sites.yml` file.

Related https://github.com/pantheon-systems/pyml-validator/issues/13

**Review by jazzsequence — APPROVED** [review]

File: source/content/guides/multisite/07-search-replace.md
Comment (pwtyler) [reply source/content/guides/multisite/07-search-replace.md:31]: actually I see `search and replace` in docs I didn't touch, going to followup with that 😁

File: source/content/guides/multisite/07-search-replace.md
Comment (pwtyler) [reply source/content/guides/multisite/07-search-replace.md:46]: Will address as part of CMSP-319 and the docs updates to come out of that.

File: source/content/guides/multisite/07-search-replace.md
Comment (pwtyler) [inline source/content/guides/multisite/07-search-replace.md]:
```suggestion
If search-replace is enabled for an environment, but `sites.yml` does not exist, nothing will be updated. if `sites.yml` is different between environments, the `domain_maps` in the target environment’s `sites.yml` will be used to determine what to replace.
```

**Review by jspellman814 — APPROVED** [review]

File: source/content/guides/multisite/07-search-replace.md
Comment (CdrMarks) [inline source/content/guides/multisite/07-search-replace.md:31]: This could be `and` or `&`, but I think it needs one of them.
```suggestion
To enable search and replace between environments on a WPMS site, add 
```

File: source/content/guides/multisite/07-search-replace.md
Comment (CdrMarks) [inline source/content/guides/multisite/07-search-replace.md:48]:
```suggestion
If search and replace is enabled for an environment, but `sites.yml` does not exist, nothing will be updated. If `sites.yml` is different between environments, the `domain_maps` in the target environment’s `sites.yml` will be used to determine what to replace.
```

File: source/content/guides/multisite/07-search-replace.md
Comment (CdrMarks) [inline source/content/guides/multisite/07-search-replace.md:50]:
```suggestion
There is currently a limit of 25 sites that can be configured for search and replace. We expect this number to increase before General Availability.
```

**Review by CdrMarks — DISMISSED** [review]
I agree with John regarding the mixed usages of:
* search-replace
* search and replace
* search & replace

**Comment by jazzsequence** [comment]
This should be good to merge unless @CdrMarks or @scottbuscemi have any objections

**Comment by scottbuscemi** [comment]
Gooooood to merge :)

**Review by whitneymeredith — APPROVED** [review]
LGTM

**Pull request merged**
